pkgs/tools/misc/watchexec/default.nix

```diff
@@ -1,17 +1,24 @@
-{ stdenv, rustPlatform, fetchFromGitHub }:
+{ stdenv, lib, rustPlatform, fetchFromGitHub, CoreServices, CoreFoundation }:
 
-rustPlatform.buildRustPackage rec {
+with rustPlatform;
+
+buildRustPackage rec {
   name = "watchexec-${version}";
   version = "1.9.0";
+  buildInputs = stdenv.lib.optional stdenv.isDarwin [ CoreServices ];
+
+  preConfigure = ''
+    export NIX_LDFLAGS="-F${CoreFoundation}/Library/Frameworks -framework CoreFoundation $NIX_LDFLAGS"
+  '';
 
   src = fetchFromGitHub {
     owner = "watchexec";
     repo = "watchexec";
     rev = version;
-    sha256 = "0zp5s2dy5zbar0virvy1izjpvvgwbz7rvjmcy6bph6rb5c4bhm70";
+    sha256 = "1jib51dbr6s1iq21inm2xfsjnz1730nyd3af1x977iqivmwdisax";
```

Comment:
Is there something in the sources that results in a different hash on darwin because of case normalisation?

Reply:
Looks like. When I try this locally on x86_64-linux (starting with an all-zeroes hash), I get the original hash "0zp5s...". So @LnL7 what do we do about it?

Reply:
I can't reproduce this hash.

```
$ nix-build -A watchexec.src --check                                                                                                                                                                                                                                /src/nixpkgs
these paths will be fetched (0.02 MiB download, 0.11 MiB unpacked):
  /nix/store/gh84q49b42kmj94290sswvgcxd648whn-source
copying path '/nix/store/gh84q49b42kmj94290sswvgcxd648whn-source' from 'https://cache.nixos.org'...
checking outputs of '/nix/store/lg5sa2bwasp43vhdxw5kaqav148gf3y6-source.drv'...

trying https://github.com/watchexec/watchexec/archive/1.9.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   126    0   126    0     0    115      0 --:--:--  0:00:01 --:--:--   115
100 30620    0 30620    0     0  12267      0 --:--:--  0:00:02 --:--:-- 25537
unpacking source archive /private/tmp/nix-build-source.drv-0/1.9.0.tar.gz
warning: rewriting hashes in '/nix/store/gh84q49b42kmj94290sswvgcxd648whn-source'; cross fingers
fixed-output derivation produced path '/nix/store/7r3rf4dz5cx81qnaiybypbp9nwzcbp2n-source' with sha256 hash '0zp5s2dy5zbar0virvy1izjpvvgwbz7rvjmcy6bph6rb5c4bhm70' instead of the expected hash '1jib51dbr6s1iq21inm2xfsjnz1730nyd3af1x977iqivmwdisax'
error: build of '/nix/store/lg5sa2bwasp43vhdxw5kaqav148gf3y6-source.drv' failed
```

Reply:
Well, then let's keep the hash unchanged at `0zp5s2dy5zbar0virvy1izjpvvgwbz7rvjmcy6bph6rb5c4bhm70` because we both get that.

Reply:
I propose we keep the hash as is, otherwise the linux build might start to fail.